lib/helpers.js

```diff
@@ -56,10 +56,14 @@ exports.prefixedHash = x => {
 
 exports.sha256 = x => crypto.createHash('sha256').update(x).digest('base64');
 
-// generates a `groups * 3` char nonce; output is a multiple of 3 to ease
-// human readability
-exports.generateNonce = async ({groups = 3} = {}) => {
-  const bytes = await randomBytes(groups * 3);
+exports.generateNonce = async ({type = 'human'} = {}) => {
+  let bytes;
+  if(type === 'human') {
+    bytes = await randomBytes(6);
+  }
+  if(type === 'machine') {
+    bytes = await randomBytes(32);
+  }
```

Comment:
For a machine, we should just use the bnid library we use elsewhere and generate a 128-bit random nonce. For humans we can use the alphabet stuff here.

Reply:
What security/usability tradeoff do we want for the human version?  Do we also need a simple digits-only version for easy mobile entry?

Reply:
With 9 characters where numeric-only digits are used we have 1B combinations. It would be only 1M for 6 chars w/numeric digits. For 6 chars with the current alphabet we get ~38B. If 1B is enough -- because we should be strongly rate limiting the options and aggressively using expiration -- then 9 characters with numeric-only digits seems like it would work and be a decent middle ground. The 9 digit code would be broken up into groups of 3 which is usually good enough for humans to remember or enter easily. @davidlehn thoughts?

Reply:
For the use case requiring ease of entry, 9 digit codes sound ok, as long as restrictions such rate limiting and expiration are in place.

Reply:
+1 ... the code should be updated to use 9 character numeric-only digits for `human` `entryStyle` and 128-bit bnid fixed values for `machine`.